src/types/profile-derived.js

```diff
@@ -202,6 +204,53 @@ export type LocalTrack =
 export type Track = GlobalTrack | LocalTrack;
 export type TrackIndex = number;
 
+/**
+ * The origins timeline view is experimental. These data structures may need to be
+ * adjusted to fit closer to the other track types, but they were easy to do for now.
+ */
+
+/**
+ * This origin was loaded as a sub-frame to another one. It will be nested in the view.
+ */
+export type OriginsTimelineEntry = {|
```

Comment:
I like this type of structure for the timeline entries. But we also have other stuff like screenshot and other visual-progress type. I wonder how we can fit these into this type of structure.

Reply:
The data structure for tracks that I initially wrote proved really really annoying to actually work with. Let's see how this one goes.